src/cc/bcc_syms.cc

```diff
@@ -547,27 +542,45 @@ int bcc_buildsymcache_resolve(void *resolver,
   return bsym->resolve_addr(build_id, trace->offset, sym) ? 0 : -1;
 }
 
-struct mod_st {
-  const char *name;
-  uint64_t start;
-  uint64_t file_offset;
-};
+int _bcc_syms_find_module(mod_info *info, int enter_ns, void *p) {
+  struct mod_search *mod = (struct mod_search *)p;
+  // use inode + dev to determine match if inode set
+  if (mod->inode) {
```

Comment:
Agree, if inode information is available, we should try inode comparison first. This will be more robust.